apiserver/common/modeluser.go

```diff
@@ -4,28 +4,32 @@
 package common
 
 import (
+	"context"
 	"time"
 
 	"github.com/juju/errors"
-	"github.com/juju/names/v5"
 
+	coremodel "github.com/juju/juju/core/model"
 	"github.com/juju/juju/core/permission"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
 )
 
-type modelConnectionAbleBackend interface {
-	LastModelConnection(names.UserTag) (time.Time, error)
+// accessService provides the methods from the needed from the access service.
+type accessService interface {
+	// LastModelConnection gets the time the specified user last connected to the
+	// model.
+	LastModelConnection(context.Context, coremodel.UUID, string) (time.Time, error)
 }
 
 // ModelUserInfo converts permission.UserAccess to params.ModelUserInfo.
-func ModelUserInfo(user permission.UserAccess, st modelConnectionAbleBackend) (params.ModelUserInfo, error) {
+func ModelUserInfo(ctx context.Context, st accessService, modelUUID string, user permission.UserAccess) (params.ModelUserInfo, error) {
```

Comment:
Can we combine the call to ReadAllUserAccessForTarget and LastModelConnection is some way? We don't need 2 calls into the same service to get this data. 

ReadAllUserAccessForTarget is replacing Users(). It was poorly named to begin with.